src/backend/data/post.js

```diff
@@ -1,41 +1,45 @@
-const { getPost, addPost } = require('./utils/storage');
-const { logger } = require('./utils/logger');
-const sanitizeHTML = require('./utils/sanitize-html');
-const textParser = require('./utils/text-parser');
+const { getPost, addPost } = require('../utils/storage');
+const { logger } = require('../utils/logger');
+const sanitizeHTML = require('../utils/sanitize-html');
+const textParser = require('../utils/text-parser');
+const hash = require('./hash');
 
 function toDate(date) {
-  // Is this already a Date?
-  if (typeof date === 'object' && typeof date.getTime === 'function') {
+  if (date instanceof Date) {
     return date;
   }
-
   return new Date(date);
 }
 
 class Post {
   constructor(author, title, html, text, datePublished, dateUpdated, postUrl, siteUrl, guid) {
+    // Use the post's guid as our unique identifier
+    this.id = hash(guid);
     this.author = author;
     this.title = title;
     this.html = html;
     this.text = text;
-    this.published = toDate(datePublished);
-    this.updated = toDate(dateUpdated);
+    this.published = datePublished ? toDate(datePublished) : new Date();
+    this.updated = dateUpdated ? toDate(dateUpdated) : new Date();
     this.url = postUrl;
     this.site = siteUrl;
     this.guid = guid;
```

Comment:
Interesting idea.  I wonder if we'd be wise to keep it around, since it comes [from the RSS/Atom article standard](https://cyber.harvard.edu/rss/rss.html#ltguidgtSubelementOfLtitemgt) (that's where the name came from here), and is the field we use to calculate the hash.  If we throw it away, we can't ever confirm/debug our hashes.  I'm inclined to leave it for now, and we can ditch it later if we find that we really don't ever use it, after our code and data backend settle a bit more.